dvc/command/daemon.py

```diff
@@ -56,12 +58,17 @@ def add_parser(subparsers, parent_parser):
     )
     daemon_updater_parser.set_defaults(func=CmdDaemonUpdater)
 
-    DAEMON_ANALYTICS_HELP = "Send dvc usage analytics."
+    DAEMON_ANALYTICS_HELP = "Collect and send dvc usage analytics."
     daemon_analytics_parser = daemon_subparsers.add_parser(
         "analytics",
         parents=[parent_parser],
         description=DAEMON_ANALYTICS_HELP,
         help=DAEMON_ANALYTICS_HELP,
     )
-    daemon_analytics_parser.add_argument("target", help="Analytics file.")
+    daemon_analytics_parser.add_argument(
+        "cmd_class", help="Class called through main"
+    )
+    daemon_analytics_parser.add_argument(
+        "ret", help="Return code from running such class"
+    )
```

Comment:
Let's not bloat CLI. It will be bad when we'll need to pass some additional info like exc class or whatnot. You could pass a json file with incomplete data and let the worker collect the rest (system and stuff like that).

Reply:
It is an internal CLI. I don't know how willing are we to changing the analytics report, but if it is something that we are considering for the future, then I agree we should be more flexible.

What about a single target, `exec_info` that contains the string representation of a JSON:
`'{"cmd_class": "CmdAdd", "cmd_return_code": 0}'` and pass it to the `daemon analytics` command, then, merging it with the actual report:

```python
report = analytics.report()
report.update(self.args.exec_info)
```

Dealing with writing, reading, and deleting a file seems like extra effort with no extra benefit.

Reply:
@mroutis Shell escaping rules differ for different systems, plus CLI arg length is limited, so that approach is dangerous.

Reply:
Indeed, it is dangerous, but it depends :sweat_smile:

https://support.microsoft.com/en-us/help/830473/command-prompt-cmd-exe-command-line-string-limitation
> On computers running Microsoft Windows XP or later, the maximum length of the string that you can use at the command prompt is 8191 characters

https://www.in-ulm.de/~mascheck/various/argmax/
`getconf ARG_MAX`

It is probably enough space to send a couple of characters.
Also, `subprocess.Popen` escape the string literals for you.

I don't mind using a file if this would put us on the same page)

Reply:
@mroutis I wouldn't really rely on it, especially when we are talking about json data. Best to just put it into the temp file and go from there.

Reply:
got it, @efiop; changed it to a tempfile)